### PR TITLE
Implement the groupInvitationsWithdraw service

### LINKED ISSUE #346: Service for invitation withdrawal
Service allowing a group manager (with can_manage>=membership on the group or one of his ancestor) to withdraw a invitation.

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #346

### CODE REVIEW

File: app/api/groups/groups.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func (srv *Service) SetRoutes(router chi.Router) {
 	router.Post("/groups/{parent_group_id}/requests/reject", service.AppHandler(srv.rejectRequests).ServeHTTP)
 
 	router.Post("/groups/{parent_group_id}/invitations", service.AppHandler(srv.inviteUsers).ServeHTTP)
+	router.Post("/groups/{parent_group_id}/invitations/withdraw", service.AppHandler(srv.withdrawInvitations).ServeHTTP)
```
Comment (smadbe) [inline app/api/groups/groups.go]: We should have done that with a DELETE.  Unlike accept/refuse, "withdraw" is really just a delete. 
(maybe not great for mass operations anyway)

File: app/api/groups/withdraw_invitations.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	"net/http"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation POST /groups/{parent_group_id}/invitations/withdraw groups users groupInvitationsWithdraw
+// ---
+// summary: Withdraw invitations to join a group
+// description:
+//   Lets an admin withdraw invitations to join a group.
```
Comment (smadbe) [inline app/api/groups/withdraw_invitations.go]: "a manager" ?

File: app/api/groups/withdraw_invitations.robustness.feature [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  Scenario: Fails when the user is a manager of the parent group, but doesn't have enough rights to manage memberships
+    Given I am the user with id "12"
+    When I send a POST request to "/groups/13/invitations/withdraw?group_ids=31,141,21,11,13"
+    Then the response code should be 403
+    And the response error message should contain "Insufficient access rights"
+    And the table "groups_groups" should stay unchanged
+    And the table "groups_ancestors" should stay unchanged
+
+  Scenario: Fails when the user doesn't exist
```
Comment (smadbe) [inline app/api/groups/withdraw_invitations.robustness.feature]: This is duty for the middleware, I don't think we should test that in each service.

File: app/api/groups/withdraw_invitations.go (hunk 2 above)
Comment (zenovich) [reply app/api/groups/withdraw_invitations.go]: fixed

File: app/api/groups/withdraw_invitations.robustness.feature (hunk 3 above)
Comment (zenovich) [reply app/api/groups/withdraw_invitations.robustness.feature]: Should I remove the test?

Comment (smadbe) [reply app/api/groups/withdraw_invitations.robustness.feature]: No, that's ok.

**Pull request merged**